### ISSUE TITLE
Differences in Terminology and Definitions in GlassBR SRS

### ISSUE THREAD

**Issue by samm82** [JacquesCarette/Drasil#835]
The following differences exist between the Drasil (left) and caseStudies "Terminology and Definitions" sections:

- [ ] Sort alphabetically - Drasil - to be revisited after #1244
- [x] Sort alphabetically - caseStudies
- [x] Add missing reference for probability of breakage in Drasil
![p_b](https://user-images.githubusercontent.com/35857611/42584575-7ee06b88-8501-11e8-870c-88b5d97e5509.png)
- [x] Add coordinates sentence to SD in caseStudies
![sd](https://user-images.githubusercontent.com/35857611/42585445-aaea4558-8503-11e8-80b3-4964f06c65d7.png)

### Glass Types
- [x] Change "Glass Types" to "Glass Type" in caseStudies
- [x] Remove "glass" to avoid the location of acronym difference - Drasil
- [x] Remove "glass" to avoid the location of acronym difference - caseStudies
- [x] Add missing commas after "psi)" in caseStudies
![glass types](https://user-images.githubusercontent.com/35857611/42584265-cba22462-8500-11e8-8f61-1ffbc1afcb2c.png)

### Load
- [x] Add definition for "load" in Drasil
- [x] Add additional reference information in caseStudies
![load](https://user-images.githubusercontent.com/35857611/42584505-51936572-8501-11e8-87f4-b6d07804e6f2.png)

**Comment by smiths** [comment]
In the future, can you do separate issues for each item.  I'm doing a lot of scrolling up and down to keep track of what issues you have identified.  I think it will save us both time in the long run, even if the issues end up repetitive in terms of the figures you attach.

I believe you are correct that in both cases the ordering of the terminology is arbitrary.  Unless you can see a reason otherwise, why don't we standardize the terminology section to list the definitions in alphabetical order.  Would this be easy to do in the Drasil version?

Please add the missing reference to the Drasil version.

Please add the extra sentence for the components of the stand off distance.

Please add the definition of load to the Drasil version.

Please add the additional reference information to the case study.

**Comment by samm82** [comment]
Perfect and sorry - I thought that issues like these should be grouped. 

Not sure how difficult alphabetical sorting would be. I did implement the sorting of references alphabetically, and it shouldn't be _too_ different.

What do you think about the proposed changes to Glass Types?

**Comment by smiths** [comment]
I prefer Glass Type to Glass Types.  Can you please make this consistent between the examples.

**Comment by samm82** [comment]
I think the commas should be added to caseStudies, but the location of acronyms might be arbitrary?

**Comment by smiths** [comment]
Sorry I missed these specific points you raised.  (That is why I like smaller issues.)  😄 

Yes, please add the commas.  As for the location of the acronyms, I agree it is arbitrary.  Why don't we delete the word glass from both cases (it is redundant anyway) and then both examples will be the same.

**Comment by samm82** [comment]
Makes sense - I'll be sure to make smaller issues in future, and sounds good

**Comment by smiths** [comment]
Okay.

**Comment by samm82** [comment]
I think the formatting might have to be a little different for the additional referencing information in caseStudies, as the reference is hardcoded in Drasil right now. Without hardcoding, I don't really think there's a way to add additional notes inside the square brackets of a reference. @smiths

**Comment by smiths** [comment]
You can add information to \cite, like \cite[chapter, p.~215]{citation01} (from https://en.wikibooks.org/wiki/LaTeX/Bibliography_Management).  Would this work?

**Comment by samm82** [comment]
Yes this works, but it adds a comma after the reference number. Not a big deal I don't think

**Comment by samm82** [comment]
Removing "Glass" from the Terminology and Definitions list also removes it from the table of abbreviations and acronyms, causing it to look like:

|Abbreviation|FullForm|
|---|---|
|AN|Annealed |

a) Is this acceptable?
b) Should this change be propagated to caseStudies? @smiths

**Comment by smiths** [comment]
Thank you for the question.  I'm fine with the consequences of the change.  Yes, we should update it in case studies too.

**Comment by samm82** [comment]
https://github.com/JacquesCarette/Drasil/blob/46967c44b13f1c82253ecd6ffa63e85cf69ee9cc/code/drasil-example/Drasil/GlassBR/Body.hs#L205-L216

The last entry (probability of breakage) should be reworked - either in order to sort, or otherwise.

**Comment by samm82** [comment]
Also, the way this is done seems like a bit of a hack - could we merge `tAndDWAcc `, `tAndDOnly`, and `tAndDWSym` into one function via pattern matching or some other method? @szymczdm @JacquesCarette 

https://github.com/JacquesCarette/Drasil/blob/46967c44b13f1c82253ecd6ffa63e85cf69ee9cc/code/drasil-example/Drasil/GlassBR/Body.hs#L223-L226

**Comment by szymczdm** [comment]
If you want to merge functions, first we need to make sure their type signatures allow it. Can you find where they're defined?

**Comment by samm82** [comment]
This might not work, as `tAndDWSym` takes both a `Concept` and a `Quantity`, but the only place in all the examples where it is used is in the above example
https://github.com/JacquesCarette/Drasil/blob/46967c44b13f1c82253ecd6ffa63e85cf69ee9cc/code/drasil-data/Data/Drasil/SentenceStructures.hs#L168-L178
https://github.com/JacquesCarette/Drasil/blob/46967c44b13f1c82253ecd6ffa63e85cf69ee9cc/code/drasil-example/Drasil/GlassBR/Body.hs#L215

So if we merged `probBreak` and `prob_br` into a single `ConceptChunk` and change how `tAndDWSym` would function within our new merged function, this could work. 

The only problem I could see with this approach would be if we used, say `tAndDWAcc` and `tAndDOnly` on the same chunk. However, then we could just make our merged function call the other three for cases like these, but we could still use the individual functions to specify how to display certain definitions at certain places in the document.

**Comment by JacquesCarette** [comment]
I am somewhat confused - I probably am just missing some context. It does feel like there is something to do here, but I'm not sure what. [Sure looks like there are a lot of display hacks going on in this code].

**Comment by samm82** [comment]
Would it be a hack to make a `probBreakCC` that is a `ConceptChunk` solely for the purpose of using the `tAndDWSym`? It would take the symbol of `prob_br` as the "abbreviation" for `probBreak`, so `tAndDWAcc` could be applied to just the new `ConceptChunk`, `tAndDWSym` can be removed (?) as wouldn't be used anymore, and then we could worry about merging the two remaining functions, `tAndDWAcc` and `tAndDOnly`.

@JacquesCarette @szymczdm

**Comment by JacquesCarette** [comment]
It is not a hack to take some information (such as `probBreakCC`) and make it into the most-informative-type-possible. It is a hack if you fill in extra fields with 'junk', just so that a particular function will work -- especially if that junk will then appear in the results. If the junk doesn't appear, then the true bug might be that the function is requesting too much structure of its input, i.e. more than it uses.

**Comment by samm82** [comment]
https://github.com/JacquesCarette/Drasil/blob/59b5f43a7e1e76038dcf75f27531a040173184ba/code/drasil-data/Data/Drasil/SentenceStructures.hs#L178-L188

Based on the comments here, I think it would make sense to treat acronyms and symbols as equivalent (ie. as short forms of the name). This would allow for easy merging of these functions. Does this sound like an appropriate fix? Not sure about the design implications of this.

**Comment by smiths** [comment]
@samm82, your proposal to treat acronyms and symbols as equivalent sounds interesting, but I have a feeling that this would be a significant change.  From scanning through this issue, it is unclear what problem you are actually trying to solve.  This is why I don't like "omnibus" issues.  There is too much going on, and the title doesn't give an idea of the problem.

My suggestion is that you create a new issue that clearly describes what problem is being solved, and then recommends the potential solution you proposed.  This will help us to have a more focused discussion.  You could then close this issue, and point to it from the new issue.  After all, it seems like almost everything in the original checklist has been checked off.

**Comment by samm82** [comment]
This issue came up because of sorting - I think the first step to sorting was to make sure the list was consistent in how it was implemented? I'll revisit this once #1244 is resolved. Should I close this issue and make a new one specifically for the sorting?

**Comment by smiths** [comment]
Yes, please create an issue specific to sorting.

**Comment by samm82** [comment]
#1245
